logging/tests/unit/handlers/transports/test_background_thread.py

```diff
@@ -69,103 +74,217 @@ def _get_target_class():
     def _make_one(self, *args, **kw):
         return self._get_target_class()(*args, **kw)
 
+    def _make_one_with_mock_thread(self, *args, **kw):
+        with mock.patch('threading.Thread', new=_Thread):
+            with mock.patch('atexit.register') as atexit_mock:
+                self.atexit_mock = atexit_mock
```

Comment:
To prevent unnecessary atexit handlers holding on to references to objects created in tests.